chain/beacon/cache_test.go

```diff
@@ -3,52 +3,56 @@ package beacon
 import (
 	"testing"
 
+	"github.com/drand/kyber/share"
 	"github.com/stretchr/testify/require"
 
 	"github.com/drand/drand/chain"
-	"github.com/drand/drand/common/scheme"
+	"github.com/drand/drand/crypto"
 	"github.com/drand/drand/key"
 	"github.com/drand/drand/log"
 	"github.com/drand/drand/protobuf/drand"
-	"github.com/drand/kyber/share"
 )
 
-var fakeKey = key.NewKeyPair("127.0.0.1:8080")
+var fakeKey, _ = key.NewKeyPair("127.0.0.1:8080", nil)
 
-func generatePartial(idx int, round uint64, prev []byte) *drand.PartialBeaconPacket {
-	sch := scheme.GetSchemeFromEnv()
-	verifier := chain.NewVerifier(sch)
+func generatePartial(t *testing.T, idx int, round uint64, prev []byte) *drand.PartialBeaconPacket {
+	t.Helper()
+	sch, err := crypto.GetSchemeFromEnv()
+	require.NoError(t, err)
 
 	sh := &share.PriShare{
 		I: idx,
 		V: fakeKey.Key,
 	}
 
-	msg := verifier.DigestMessage(round, prev)
-	sig, _ := key.Scheme.Sign(sh, msg)
-	return &drand.PartialBeaconPacket{
-		Round:       round,
-		PreviousSig: prev,
-		PartialSig:  sig,
+	partialBeacon := &drand.PartialBeaconPacket{
+		Round:             round,
+		PreviousSignature: prev,
 	}
+
+	msg := sch.DigestBeacon(partialBeacon)
+	partialBeacon.PartialSig, err = sch.ThresholdScheme.Sign(sh, msg)
```

Comment:
Not a deal breaker (still LGTM i think the PR) but this could be inside the scheme given it's a crypto operation. SO `scheme.GeneratePartialSig(partialBeacon)`

Reply:
This is done in the `func (v *Vault) SignPartial` function I'd say. Scheme doesn't hold the private keys/shares, but the `Vault` does. 
This is in test code, so I'll let as is for now.